smac_planner/src/smac_planner.cpp

```diff
@@ -239,22 +265,32 @@ nav_msgs::msg::Path SmacPlanner::createPlan(
   steady_clock::time_point a = steady_clock::now();
 #endif
 
+  // Choose which costmap to use for the planning
+  std::shared_ptr<nav2_costmap_2d::Costmap2D> costmap;
+  if (shouldDownsample()) {
+    _costmap_downsampler->downsample();
+    _costmap_pub->publishCostmap();
+    costmap = _downsampled_costmap;
+  } else {
+    costmap = std::make_shared<nav2_costmap_2d::Costmap2D>(*_costmap);
```

Comment:
You shouldn't need this copy. Is there a way you can make the costmaps the same pointer type or `auto` the `costmap` so that it can use either a raw or smart pointer? This copy to deference the costmap and copy into a shared pointer will be heavy.

Reply:
It seems like we control the _downsampled_costmap type, so we can control the `costmap` type. Maybe have `costmap` be a raw ptr (or whatever `_costmap` is) and then do `_downsampled_costmap.get()` if we can't make it outright the same